packages/core/src/operators/matchPath/matchPath.operator.ts

```diff
@@ -11,19 +11,22 @@ type MatcherOpts = {
   combiner?: boolean;
 };
 
-export const matchPath = (path: string, opts: MatcherOpts = {}) => (
-  source$: Observable<HttpRequest>
-) =>
+const isRequestMatched = (req: HttpRequest) =>
+  !(req.matchPath && req.matchType);
+
+export const matchPath = (path: string, opts: MatcherOpts = {}) => (source$: Observable<HttpRequest>) =>
   source$.pipe(
     tap(req => (req.query = req.query || {})),
     tap(req => (req.params = req.params || {})),
     tap(req => (req.matchers = req.matchers || [])),
+    filter(isRequestMatched),
     filter(req => {
       const matcher = matcherFactory(req.matchers!, path, opts.suffix);
       const url = removeQueryParams(req.url!);
       return pathToRegexp(matcher).test(url);
     }),
     tap(req => (req.params = urlParamsFactory(req, path))),
     tap(req => (req.query = queryParamsFactory(req.url!))),
-    tap(req => opts.combiner && req.matchers!.push(path))
+    tap(req => opts.combiner && req.matchers!.push(path)),
+    tap(req => (req.matchPath = !opts.combiner ? true : false)),
```

Comment:
`req.matchpath = !opts.combiner` -> I guess this will be the same :)

Reply:
IMO current implementation is more clear